### ISSUE TITLE
Add documentation on how to update to newer versions

### ISSUE THREAD

**Issue by skrysmanski** [dotnet/roslyn-analyzers#4506]
Version 5.0.1 has just shipped and I can't figure out how to get this version into my projects.

I've tried updating `<AnalysisLevel>` property in my project file - however I can't verify that the newer version is actually being used. Visual Studio still shows "5.0". In fact, even if I set the property to the non-existing version of "5.0.8" in my project file everything still builds fine - I don't get any errors (which I would expect).

I guess I could switch to NuGet but then the README says:

> You do not need to manually install this NuGet package to your project if you are using .NET5 SDK or later. These analyzers are enabled by default for projects targeting .NET5 or later.

So, I think the documentation in the main README should be improved on how to get newer versions.

**Comment by Youssef1313** [comment]
AFAIK, you only need to update the .NET SDK to 5.0.1, and you'll get the analyzers updated with it.
To ensure that every developer has the specified SDK version, you can use a `global.json`, see https://github.com/dotnet/roslyn-analyzers/blob/master/global.json as an example, and see https://docs.microsoft.com/dotnet/core/tools/global-json for documentation on global.json.

For the part of AnalysisLevel, I think an error message makes sense if possible.

**Comment by skrysmanski** [comment]
@Youssef1313 

> AFAIK, you only need to update the .NET SDK to 5.0.1, and you'll get the analyzers updated with it.

I guess this will happen automatically with a future update of Visual Studio (on Windows)?

**Comment by mavasani** [comment]
Version 5.0.1 of NetAnalyzers was only released to match the FxCopAnalyzers package 3.3.2 for seamless transition from the latter to former. .NET SDK analyzer version will eventually catch up, but cadence of updates to NuGet released on NuGet.org will be faster then cadence of updates of analyzer assemblies shipping inside the .NET SDK.

**Comment by skrysmanski** [comment]
So, it's still a valid way to get the .NET analyzers via NuGet (instead of using the ones bundled with the .NET SDK) like before, right? If so, I think this should be made more clear.

**Comment by Youssef1313** [comment]
@skrysmanski Yes it's valid to get the analyzers via the NuGet package. See <https://docs.microsoft.com/dotnet/fundamentals/code-analysis/overview#latest-updates>.

**Comment by loraderon** [comment]
After experiencing the issue reported in #4755 I noticed that you can force a newer version using `<AnalysisLevel>5.0.3</AnalysisLevel>`.

`<AnalysisLevel>latest</AnalysisLevel>` **does not** use the latest released version.
